### PR TITLE
2141 colin getcorpnum

### PR DESCRIPTION
**Pull request by peter-freshworks**
*Issue #:* /bcgov/entity#2141

*Description of changes:*
-Get the next Corp Number in series for entity type
-Initial incorporation (including offices) from NR JSON

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by kialj876** [comment]
1 flake8 error that needs to be fixed!

**Comment by kialj876** [comment]
The reset methods are failing too and the post incorporation test via postman. Do these work for you locally?

File: colin-api/src/colin_api/models/reset.py [hunk 1]
```diff
@@ -86,6 +86,22 @@ def get_filings_for_reset(self):
             current_app.logger.error('error getting filing/event info for reset for: {}'.format(self.as_dict()))
             raise err
 
+    @classmethod
+    def get_incorporations_by_event(cls, cursor, event_ids: list):
```
Comment (kialj876) [inline colin-api/src/colin_api/models/reset.py]: Move this to after the protected methods pls (for better readability).

Comment (peter-freshworks) [reply colin-api/src/colin_api/models/reset.py]: this actually should be a protected method, I have moved it to the last protected method

File: colin-api/src/colin_api/models/filing.py [hunk 2]
```diff
@@ -691,7 +715,7 @@ def add_filing(cls, con, filing):  # pylint: disable=too-many-locals,too-many-st
         """
         try:
             corp_num = filing.get_corp_num()
-            user_id = 'COOPER' if corp_num[:2] == 'CP' else None
+            user_id = 'COOPER' if corp_num[:2] in ('CP', 'BC') else None
```
[the remark this answers is not in the document]
Comment (peter-freshworks) [reply colin-api/src/colin_api/models/filing.py]: Discussed with Bob, BCOMPS use a different user id ("BCOMPS")

File: colin-api/src/colin_api/models/business.py [hunk 3]
```diff
@@ -339,3 +348,95 @@ def reset_corp_states(cls, cursor, event_ids: list):
         except Exception as err:
             current_app.logger.error(f'Error in Business: Failed reset ended corp_state rows for events {event_ids}')
             raise err
+
+    @classmethod
+    def get_next_corp_num(cls, corp_type):
```
[the remark this answers is not in the document]
Comment (kialj876) [reply colin-api/src/colin_api/models/business.py]: Oh actually I'm getting one failing pytest on the test_registered_office

[the remark this answers is not in the document]
Comment (kialj876) [reply colin-api/src/colin_api/models/business.py]: I'm also getting errors on the reset methods: ERROR in reset:reset.py:143 - _delete_corp_name: Error in Reset: failed to delete from corp_name table.

[the remark this answers is not in the document]
Comment (kialj876) [reply colin-api/src/colin_api/models/business.py]: Both POST calls in the pm tests for incorporation also fail for me. Looks like a trigger problem though. Have you looked into what's causing that and whether or not we can drop that trigger from oracle-dev?

**Pull request merged**